hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -2431,11 +2431,35 @@ private static boolean isCatalogTable(final TableName tableName) {
     return tableName.equals(TableName.META_TABLE_NAME);
   }
 
+  private void checkSnapshot(TableName tableName, boolean archive) throws IOException {
+    /*
+     * If decide to delete the table without archive, need to make sure the table has no snapshot
+     * Meanwhile, the check will scan the snapshots which will do list and open, if there is lots of
+     * snapshot, the performance may be impacted, should evaluate the performance between directly
+     * archive and snapshot scan TODO: find some any way to get if the table snapshotted or not
+     */
+    if (!archive) {
```

Comment:
Why we need this logic here? Mind exlaining a bit?

Reply:
@Apache9 Thanks so much for your time. The thought is because there will be no archive for all the hfiles, so if there are any snapshots, the snapshots will be broken, so to avoid the broken, I made the check before conduct the actual delete without archive, make sense?

Reply:
Ah, this is a problem...

I think the correct way is to still archive the files which are referenced by snapshots, and delete unreferenced files directly? This wil be a bit hard as snapshot can happen at the same time?

Reply:
@Apache9 You are correct, there was some compromise for the performance and accuracy. If need to scan all the snapshot files and find out if the hfile referred, there will be another performance impact, meanwhile, for the snapshot which is happening, we may need a get the lock from the snapshot manager to make sure no in processing snapshot referring the file, its another concern.

So current I just simplify the logic and return the message to end user and let user decide if they want to remove the snapshot and then delete the table without archive. Do you think its acceptable? Or we had better check the details with some performance sacrifice?